src/xtgeo/surface/_regsurf_import.py

```diff
@@ -250,13 +251,46 @@ def _import_irap_ascii(mfile):
     return args
 
 
-def import_ijxyz(mfile, template=None, **_):
+def import_ijxyz(mfile, template=None, engine="python", **kwargs):
     """Import OW/DSG IJXYZ ascii format."""
-    return _import_ijxyz_tmpl(mfile, template) if template else _import_ijxyz(mfile)
 
+    # override engine for this format as cxtgeo read will fail for memstreams, and the
+    # python implementation here is faster and more robust
+    use_engine = "python"
+    logger.debug("Override engine for ijxyx format; set to python")
 
-def _import_ijxyz(mfile):
-    """Import OW/DSG IJXYZ ascii format."""
+    if use_engine == "cxtgeo":  # will not be invoked in practice; remove in xtgeo 4
+        return (
+            _import_ijxyz_cxtgeo_tmpl(mfile, template)
+            if template
+            else _import_ijxyz_cxtgeo(mfile)
+        )
+
+    return _import_ijxyz_python(mfile, template)
+
+
+def _import_ijxyz_python(mfile, template):
+    """Import using pure python."""
+    # This is actually faster and more robust that using ctxgeo
+    ijxyz_data = parse_ijxyz(mfile, template)
+
+    return {
+        "ncol": ijxyz_data.ncol,
+        "nrow": ijxyz_data.nrow,
+        "xori": ijxyz_data.xori,
+        "yori": ijxyz_data.yori,
+        "xinc": ijxyz_data.xinc,
+        "yinc": ijxyz_data.yinc,
+        "values": ijxyz_data.values,
+        "ilines": ijxyz_data.ilines,
+        "xlines": ijxyz_data.xlines,
+        "yflip": ijxyz_data.yflip,
+        "rotation": ijxyz_data.rotation,
+    }
+
+
+def _import_ijxyz_cxtgeo(mfile):
```

Comment:
```suggestion

def _import_ijxyz_cxtgeo(mfile: _XTGeoFIle) -> dict[str, object]:
```

Reply:
I removed the whole function (including the C part)